Adding ppx_relit 0.1 and relit_helper 0.1

charlesetc opened the pull request:
Hi — I'm hoping to expose these two packages, both stored in the same .tar at the same github url. Do you think this will work?
Thanks very much!
Charles

mseri commented on packages/ppx_relit/ppx_relit.0.1/opam:
```diff
@@ -0,0 +1,23 @@
+license: "MIT"
+dev-repo: "https://github.com/cyrus-/relit"
+build: [
+  ["jbuilder" "build"]
```
Yous should add `”-p”  name` To make sure it is building only the package itself. Also the install and remove will be automatically generated by jbuilder, so you should delete them below

mseri replied:
This will allow you to also drop the dependency on ocamlfind that seems to be used only to remove the package

mseri commented on packages/relit_helper/relit_helper.0.1/opam:
```diff
@@ -0,0 +1,22 @@
+license: "MIT"
+dev-repo: "https://github.com/cyrus-/relit"
+build: [
+  ["jbuilder" "build"]
```
See the comments on the other opam file

mseri commented on packages/ppx_relit/ppx_relit.0.1/opam:
```diff
@@ -0,0 +1,23 @@
+]
+install: ["jbuilder" "install" "ppx_relit"]
+remove: ["ocamlfind" "remove" "project"]
+depends: "ocamlfind"
```
This depends at least on jbuilder, but looking at the jbuild file there are a lot of dependencies missing: 
https://github.com/cyrus-/relit/blob/master/ppx_relit/jbuild

mseri commented on packages/relit_helper/relit_helper.0.1/opam:
```diff
@@ -0,0 +1,22 @@
+]
+install: ["jbuilder" "install" "relit_helper"]
+remove: ["ocamlfind" "remove" "project"]
+depends: "ocamlfind"
```
Also this package needs to fix the dependencies

mseri commented:
Thanks. Can you do the cleanup suggested in the comments? To fix the dependencies you could move to dune instead, the latest version is able to generate the correct list for you

charlesetc commented:
Thanks very much! I'll address your comments and move to the latest dune

camelus commented:
##### :cloud_with_lightning: opam-lint errors <small>ffc5b25f6940ef720b629ddf65280a1d978645fb</small>

* **`ppx_relit.0.1`** has errors:

  * **error 2:** File format error at line 19, column 5: Parse error



* **`relit_helper.0.1`** has errors:

  * **error 2:** File format error at line 19, column 5: Parse error



---


##### :sunny: Installability check (9688 &rarr; 9688)

charlesetc commented:
I think I've added the appropriate dependencies: thanks again! I've also added a relit-reason package which is a modified version of reason that supports the syntax relit uses. Have I done this properly?

mseri commented:
Thanks. Looks good! Let’s wait for the CI to finish, but it should be fine now

charlesetc commented:
:disappointed: looks like the last check failed. Do you know why or how I could debug that?

charlesetc commented:
Looks like it's failing since bumping to opam-version 2.0, it doesn't have m4 available? Is there a way I can require that as a dependency?

avsm commented:
The CI is currently very broken, but a fix has been deployed and is rebuilding :)

avsm commented on packages/ppx_relit/ppx_relit.0.1/opam:
```diff
@@ -0,0 +1,30 @@
+  "base64"
+  "ppxlib"
+  "base-unix"
+  "ocamlbuild"
```
does this still depend on ocamlbuild? Looks like it's been ported to Dune

charlesetc replied:
Ah good point, we use ocamlbuild for some tests but it shouldn't be needed to build/use it.

mseri replied:
If you use it in tests you can still mark it as a test dependency by adding {test} next to it

charlesetc commented:
Do you know if this will be fixed by tomorrow? Totally understand if not; it'd be good to get a sense of how we should plan things.

mseri commented:
This is a new package, the ci is not fundamental, let’s ignore it for the moment. There is still something wrong, the travis builds are failing with

```
#=== ERROR while compiling relit-reason.0.0.1 =================================#
# context     2.0.0 | linux/x86_64 | ocaml-base-compiler.4.05.0 | file:///home/travis/build/ocaml/opam-repository
# path        ~/.opam/4.05.0/.opam-switch/build/relit-reason.0.0.1
# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build -p relit-reason -j 1
# exit-code   1
# env-file    ~/.opam/log/relit-reason-11093-075801.env
# output-file ~/.opam/log/relit-reason-11093-075801.out
### output ###
# Error: I don't know about package relit-reason (passed through --only-packages/--release)

```

charlesetc commented:
Ah good catch, thanks! I'm hoping I've fixed it but am not sure: where did you see that error?

charlesetc commented:
Okay well it looks like the packages are now being found and unpackaged and it's a different type error for each ocaml version. I'll do some thinking and see if this can be fixed. Thanks for your help!

mseri commented:
I had a brief look at the errors, ocaml 4.05 and 4.06 seem to be similar. I think fixing those and ocaml 4.07 may be easier. On the other hand I think you need 4.02 if you want to be compatible with bucklescript, right? Maybe in this case using multiple dune workspaces targeting all the compilers, and cppo to specialize some fixes could help, but it is going to be quite an amout of work.

Given that this is the first release, if you need it in time for ICFP, maybe you can fix it for one specific compiler, and ask for that in the depends bounds. Then you will have more time to fix it for all compilers, but you will have the package available and tested for the compiler version you have chosen.

mseri commented on packages/ppx_relit/ppx_relit.0.1/opam:
```diff
@@ -22,10 +22,11 @@ depends: [
   "ppxlib"
   "base-unix"
   "ocamlbuild" {test}
+  "ocaml"      {= "4.06.1"}
```
I think you can relax kt to >= 4.06.0 & < 4.07.0 but I am not picky about that

charlesetc commented:
It seems reason doesn't support 4.07 yet and 4.02 is so long ago that 4.06 seems like a good compromise for now. Thanks again for your help! :tada:

mseri commented:
Thank you!
